chains/evm/calls/contracts/deposit/deposit.go

```diff
@@ -29,3 +48,12 @@ func ConstructGenericDepositData(metadata []byte) []byte {
 	data = append(data, metadata...)
 	return data
 }
+
+func ConstructGenericDepositDataWithPriority(metadata []byte, priority uint8) []byte {
```

Comment:
This construction will not work, because there is a specific length for generic metadata, so the event handler will not be able to parse it accurately... This is something I have not foreseen. THe only way is to add some specific technical flag to separate Metadata from deposit data. 
But for now lets just remove Generic with priority use case